Marlin/src/lcd/menu/menu.cpp

```diff
@@ -452,7 +452,11 @@ bool ui_selection; // = false
 void set_ui_selection(const bool sel) { ui_selection = sel; }
 void do_select_screen(PGM_P const yes, PGM_P const no, selectFunc_t yesFunc, selectFunc_t noFunc, PGM_P const pref, const char * const string/*=nullptr*/, PGM_P const suff/*=nullptr*/) {
   if (ui.encoderPosition) {
-    ui_selection = ((ENCODERBASE) > 0) == (int16_t(ui.encoderPosition) > 0);
+    #if ENABLED(REVERSE_ENCODER_DIRECTION)
+      ui_selection = ((ENCODERBASE) > 0) != (int16_t(ui.encoderPosition) > 0);
+    #else
+      ui_selection = ((ENCODERBASE) > 0) == (int16_t(ui.encoderPosition) > 0);
+    #endif
```

Comment:
Um…
```cpp
    #if ENABLED(REVERSE_ENCODER_DIRECTION)
      #define ENCODERBASE -1
    #else
      #define ENCODERBASE +1
    #endif
```
………
```cpp
    #if ENABLED(REVERSE_ENCODER_DIRECTION)
      ui_selection = false != (int16_t(ui.encoderPosition) > 0);
    #else
      ui_selection = true == (int16_t(ui.encoderPosition) > 0);
    #endif
```
………
```cpp
    #if ENABLED(REVERSE_ENCODER_DIRECTION)
      ui_selection = int16_t(ui.encoderPosition) > 0;
    #else
      ui_selection = int16_t(ui.encoderPosition) > 0;
    #endif
```
And finally…
```cpp
      ui_selection = int16_t(ui.encoderPosition) > 0;
```

Reply:
I do believe you're on the right track in a sense. We're just going back to the original assumption that clockwise rotation of the knob is the "larger" (true) value. But then the issue that my change was meant to address needs to be revisited. I understood that the person had reversed the encoder direction, and this gave them properly increasing values on clockwise motion when editing, but when on the selection screen they didn't get the right-side option when rotating the knob clockwise.

Given that the original logic —which this PR resurrects— is the same as what we use for the FR percentage on the main screen, and the user did not mention inconsistency there, I am wondering if they expected the LEFT option to come up on clockwise motion… because their knob is above the screen…? I don't know. But I do believe that reverting to the previous code is the right idea for now.

Reply:
…but also keeping my encoder-direction cleanup from the other PR…